### PR TITLE
week-07-day-01-drill

### PR DESCRIPTION
**Pull request by ben-manning**
I've got the initial notes and solutions from WDI 19-20.  Naturally, Brianna does a great job here of describing the concepts so I don't want to go away from that.  I need to import a few things into this before I'm ready for a total merge.

### CODE REVIEW

**Comment by ben-manning** [comment]
I think I've got a fully fleshed out lesson here.  I'm going to walk through what a tree is and then have the class get into pairs to whiteboard and work through the challenges.  At the end I'll show one solution on the trees.  Let me know what y'all think.

File: week-07-rails-continued/day-01/drill/readme.md [hunk 1]
```diff
@@ -1 +1,110 @@
...
+
+### Challenges - Binary Search Trees
+
+1. Create a binary search tree from the following array: [0,1,2,3,4,5,6].
+
+2. Describe an algorithm to check if a particular number value is inside a binary search tree.  *Hint: start by checking if it's the value of the root.*
+
+3. In a binary search tree, how can you find the minimum element? The maximum?
+
+4. Assume for the following challenges that you have a binary search tree data structure. The structure will be recursive, meaning each node is actually represented as a full subtree, an instance of `BinarySearchTree`.
```
Comment (cameronjacoby) [inline week-07-rails-continued/day-01/drill/readme.md]: @ben-manning The solution for this challenge is about hex colors - is that what this challenge should be? They don't quite match up.

**Comment by cameronjacoby** [comment]
@ben-manning This looks awesome!! Just one small comment on challenge #4. Also, can you use GitHub's image uploader for the images instead of storing them inside our modules repo? You can just upload them onto the open issue for this drill, then copy the GitHub-hosted link.

**Comment by ben-manning** [comment]
trie
bst
paths
tree_terms

**Comment by ben-manning** [comment]
@cameronjacoby I've got the files cleaned up and knocked out the discrepancies from question four.  I think it's good to go.

**Pull request merged**